skimage/filters/rank/generic.py

```diff
@@ -146,6 +150,57 @@ def _preprocess_input(image, selem=None, out=None, mask=None, out_dtype=None,
     return image, selem, out, mask, n_bins
 
 
+def _handle_input_3D(image, selem, out, mask, out_dtype=None, pixel_size=1):
```

Comment:
Can you add a docstring here that explains all the checks that happen in this function?

Reply:
Sure. @ryanlu41 ?

Reply:
Thank you, and again, apologies that my review is holding you to a higher standard than existing code, which might be surprising! The library has evolved a lot since this code was written, including having many more users and contributors!

Reply:
fixed